ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -816,6 +819,11 @@ object Trees {
     val tpe = NoType
   }
 
+  case class TopLevelFieldExportDef(fullName: String, field: Ident)(
+      implicit val pos: Position) extends Tree {
+    val tpe = NoType
+  }
+
```

Comment:
Yes, I realized `TopLevelExportDef` didn't fit the bill for fields. Fields are a bit special because they need to be defined as proper Scala entities, and then they are exported. Where as top-level methods there is no Scala def to be aliased as an export. I think they're bound to be treated very differently, and so deserve separate IR nodes. I don't think we can consolidate the two `TopLevelExport` trees in one unique concept. You can't define a method eligible for export as a Scala entity first, because Scala entities cannot have `...rest` parameters, and for good reasons.

I guess in the 1.0.0 cleanup of `Trees.scala`, we'll simply have to rename `TopLevelExportDef` to `TopLevelMethodExportDef` and have it take a `MethodDef` rather than an arbitrary `Tree`.